app/ui/browser-modern/views/browser/tabbar/tab-contents.jsx

```diff
@@ -127,8 +128,14 @@ TabContents.propTypes = {
 function mapStateToProps(state, ownProps) {
   const page = PagesSelectors.getPageById(state, ownProps.pageId);
   const pageIsPinned = PagesSelectors.getPagePinned(state, ownProps.pageId);
+  let title = page.title || page.meta.title || page.location || 'Loading...';
+
```

Comment:
So if I write a page:

```<html>
  <head>
    <title>tofino://historyrestore is super buggy</title>
    …
```

it'll never show a title. (And I haven't checked, but if the title is the empty string, does JS regard that as false, and fall through?)

I think you want logic like this instead:

```
let title = page.title || page.meta.title;
if (title === undefined) {
    if (page.location && !isHistoryRestoreURL(page.location)) {
        title = page.location;
    } else {
        title = 'Loading…';
    }
}
```